fix: pagination odd number of entries

ryanwolhuter opened the pull request:
### Motivation

Having an odd number of entries revealed a mistake in the inputs used to calculate the new page number when updating the number of entries.

[**See the results of the regression test**](https://63d101d13121a53332f6218c-apwcqfpduw.chromatic.com/?path=/story/stories-pagination--odd-number-of-entries-regression-test) (go to the "interactions" panel at the bottom)

<img width="745" alt="Screenshot 2023-03-23 at 14 52 38" src="https://user-images.githubusercontent.com/39741965/227210278-c6cebf7b-00c1-4356-843f-948017a155d9.png">

### Changes

* add regression test
* use number of entries to calculate page number

daywiss reviewed (commented):
i'm having an issue when testing in storybook, even when clicking out of the regression test, when i start at "show 10, page 1", then go to show 20 or 50, it jumps to the last page

ryanwolhuter commented:
@daywiss I think I've got this fixed now.

I've set up different stories with even and odd numbers of entries that you can play with. You can also change the number of entries for the story dynamically in the controls field to check with other numbers.

[Even number of entries](https://63d101d13121a53332f6218c-amtkozqhes.chromatic.com/?path=/story/stories-pagination--even-number-of-entries)

[Odd number of entries](https://63d101d13121a53332f6218c-amtkozqhes.chromatic.com/?path=/story/stories-pagination--odd-number-of-entries)

<img width="566" alt="Screenshot 2023-03-24 at 09 53 45" src="https://user-images.githubusercontent.com/39741965/227458814-e5916d7f-d467-49b4-af59-6335ca1c9b6d.png">

I've also isolated the bug that was happening, fixed it, and updated the two regression tests to make sure it doesn't happen anymore on even or odd numbers of pages.

[Even regression test](https://63d101d13121a53332f6218c-amtkozqhes.chromatic.com/?path=/story/stories-pagination--even-number-of-entries-regression-test)

[Odd regression test](https://63d101d13121a53332f6218c-amtkozqhes.chromatic.com/?path=/story/stories-pagination--odd-number-of-entries-regression-test)

daywiss reviewed (approved):
awesome, working for me, could not break it this time
